### PR TITLE
mappings and which-key

### LINKED ISSUE #781: Align which-key and mappings.lua configuration
**Is your feature request related to a problem? Please describe.**
Hello,

I was trying to add some mappings to `mappings.lua` in this format:

```lua
		["<leader>"] = {
			x = {
				e = {
					function()
						require("dapui").eval()
					end,
					desc = "Evaluate Line",
				},
			},
		},
```

After much digging, It seems like this is supported in only `which-key` registration.

**Describe the solution you'd like**

Keep the same syntax support for both `mappings.lua` and `which-key` or only support one style. Two is just confusing and probably more maintenance. 

**Additional context**

### PR DESCRIPTION
**Pull request by osamuaoki**
This addresses issues discussed at #781

Signed-off-by: Osamu Aoki <osamu@debian.org>

Resolves #781

### CODE REVIEW

File: lua/user_example/init.lua [hunk 1]
```diff
@@ -216,16 +201,59 @@ local config = {
...
+      ["<leader>bd"] = { "<cmd>BufferLineSortByDirectory<cr>", desc = "Sort by directory" },
+      ["<leader>bD"] = { "<cmd>BufferLineSortByRelativeDirectory<cr>", desc = "Sort by rel. dir." },
+      ["<leader>be"] = { "<cmd>BufferLineSortByExtension<cr>", desc = "Sort by extention" },
+      ["<leader>bf"] = { "<cmd>Telescope buffers<cr>", desc = "Find" },
+      ["<leader>bj"] = { "<cmd>BufferLinePick<cr>", desc = "Pick to jump" },
+      ["<leader>bl"] = { "<cmd>BufferLineCloseLeft<cr>", desc = "Close left-side" },
+      ["<leader>bn"] = { "<cmd>BufferLineCycleNext<cr>", desc = "Next" },
+      ["<leader>bp"] = { "<cmd>BufferLineCyclePrev<cr>", desc = "Previous" },
+      ["<leader>br"] = { "<cmd>BufferLineCloseRight<cr>", desc = "Close right-side" },
+      ["<leader>bt"] = { "<cmd>BufferLineSortByTabs<cr>", desc = "Sort by tabs" },
```
Comment (mehalter) [inline lua/user_example/init.lua:226]: can We remove these and only leave a few examples in the user_example commented out.

File: lua/user_example/init.lua [hunk 2]
```diff
@@ -216,16 +201,59 @@ local config = {
...
     n = {
       -- second key is the lefthand side of the map
-      ["<C-s>"] = { ":w!<cr>", desc = "Save File" },
+      -- Terminal
+      -- Install ncdu package: (Disk size w/o snapshot btrfs subvolumes and .git)
+      ["<leader>tu"] = { function() astronvim.toggle_term_cmd "ncdu -x --exclude .git" end, desc = "Term for NCDU w/o junks" },
+      -- Install mc package:
+      ["<leader>tm"] = { function() astronvim.toggle_term_cmd "mc . ~" end, desc = "Term for MC" },
+      -- Install libreply-perl + libterm-readline-gnu-perl packages:
+      ["<leader>tp"] = { function() astronvim.toggle_term_cmd "reply" end, desc = "Term for perl5 REPL" },
```
Comment (mehalter) [inline lua/user_example/init.lua:213]: could we remove thse and leave the `tu` example for a lua function example?

**Review by mehalter — CHANGES_REQUESTED** [review]
Thanks for the contribution! This is a great addition to the `user_example` documentation. I have included some comments to remove some of these examples that don't add anything except repetition. I really like the additional comments and explanations :) Just want to keep the provided `user_example` as relatively minimal.

File: lua/user_example/init.lua (hunk 2 above)
Comment (osamuaoki) [reply lua/user_example/init.lua]: Yes

File: lua/user_example/init.lua (hunk 1 above)
Comment (osamuaoki) [reply lua/user_example/init.lua]: Yes

**Comment by osamuaoki** [comment]
I think Buffer example with less binding is needed here.  What do you think updated one.

**Review by mehalter — CHANGES_REQUESTED** [review]
Looks great! Could you please squash these commits to a single commit and reword the commit message to fit the conventional commit standard? Probably something like `chore(user_example): make mappings example more clear`

**Comment by mehalter** [comment]
@osamuaoki have you had a chance to update the commit message on this one?

**Comment by osamuaoki** [comment]
I think mappings set by which-key is OK.
But use of mappings table is not for speed but for robustness.  Its much simpler.

**Review by mehalter — APPROVED** [review]
LGTM

**Pull request merged**